### PR TITLE
feat: extend interface for controller specific tasks

### LINKED ISSUE #140: feat: extend interface for controller specific tasks
* extend DataObject
* add ComSglLayout
* add RMS Time simulation comfort function
* add template filter

### PR DESCRIPTION
**Pull request by SebastianDD**
Fixes #140

### CODE REVIEW

File: powerfactory_tools/interface.py [hunk 1]
```diff
@@ -1404,6 +1541,35 @@ def unit_conversion_settings(self) -> Sequence[PFTypes.UnitConversionSetting]:
...
+            class_name=PFClassId.VARIABLE_MONITOR.value,
+            location=result,
+            data=data,
+            force=force,
+            update=update,
+        )
+        variable_monitor = t.cast("PFTypes.VariableMonitor", obj) if obj is not None else None
+        # add variables to monitor
+        if variable_monitor is not None:
+            # specify the elment to monitor
```
Comment (sasanjac) [inline powerfactory_tools/interface.py:1566]: typo

File: powerfactory_tools/interface.py [hunk 2]
```diff
@@ -1495,7 +1661,10 @@ def create_study_case(
         if scenario is not None:
             self.activate_scenario(scenario)
 
-        self.switch_study_case(current_study_case.loc_name)
+        if current_study_case is not None:
+            self.switch_study_case(current_study_case.loc_name)
+        else:
+            self.deactivate_study_case(study_case)
```
Comment (sasanjac) [inline powerfactory_tools/interface.py:1667]: why deactivate?

File: powerfactory_tools/powerfactory_types.py
Comment (sasanjac) [inline powerfactory_tools/powerfactory_types.py:1]: i think we should keep enums

File: powerfactory_tools/utils/io.py
Comment (sasanjac) [inline powerfactory_tools/utils/io.py:1]: whats this for?

**Review by sasanjac — COMMENTED** [review]
Lets discuss some aspects and revise them

File: powerfactory_tools/utils/io.py
Comment (SebastianDD) [reply powerfactory_tools/utils/io.py:1]: Auxiliary functions for custom export of data structures to a preferred file type.
If user does not use the PowerFactory internal export, but want to export collected data.

File: powerfactory_tools/interface.py (hunk 2 above)
Comment (SebastianDD) [reply powerfactory_tools/interface.py:1667]: The goal is to create and not to also activate the new study case.
--> We need to temporaly activate the newly created study case to activate grids, scenarios and variants if specified by the user. After this setup, we switch back to the formerly activated study case or just deactivate the study case created.

**Review by sasanjac — APPROVED** [review]

**Pull request merged**